src/Raven.Server/Documents/Sharding/Subscriptions/SubscriptionConnectionsStateOrchestrator.cs

```diff
@@ -156,11 +172,11 @@ public override async Task HandleConnectionExceptionAsync(OrchestratedSubscripti
 
     public override void Dispose()
     {
-        DisposeWorkers();
+        DisposeWorkers(waitForSubscriptionTask: false);
         base.Dispose();
     }
 
-    public void DisposeWorkers()
+    public void DisposeWorkers(bool waitForSubscriptionTask)
```

Comment:
I think we should make this `DisposeWorkersAsync`, this waiting for the subscription task in a blocking way

Reply:
done